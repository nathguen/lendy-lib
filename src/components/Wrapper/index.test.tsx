import { render, screen } from '@testing-library/react-native';
import React from 'react';

import { Wrapper, Text } from '../';

describe('Wrapper', () => {
  it('renders Wrapper', () => {
    const container = render(<Wrapper />);

    expect(container).toMatchSnapshot();
  });

  it('renders Wrapper with children', () => {
    const container = render(
      <Wrapper>
        <Text>Test</Text>
      </Wrapper>
    );

    expect(screen.getByText('Test')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders Wrapper with baseHorizontal', () => {
    const container = render(<Wrapper baseHorizontal />);

    expect(container).toMatchSnapshot();
  });

  it('renders Wrapper with baseVertical', () => {
    const container = render(<Wrapper baseVertical />);

    expect(container).toMatchSnapshot();
  });

  it('renders Wrapper with baseHorizontal and baseVertical', () => {
    const container = render(<Wrapper base />);

    expect(container).toMatchSnapshot();
  });

  it('renders Wrapper with screen', () => {
    const container = render(<Wrapper screen />);

    expect(container).toMatchSnapshot();
  });

  describe('flex options', () => {
    it('renders Wrapper with NO row', () => {
      const container = render(
        <Wrapper>
          <Wrapper baseHorizontal>
            <Text>Item 1</Text>
          </Wrapper>

          <Wrapper baseHorizontal>
            <Text>Item 2</Text>
          </Wrapper>
        </Wrapper>
      );

      expect(container).toMatchSnapshot();
    });

    it('renders Wrapper with row', () => {
      const container = render(
        <Wrapper row>
          <Wrapper baseHorizontal>
            <Text>Item 1</Text>
          </Wrapper>

          <Wrapper baseHorizontal>
            <Text>Item 2</Text>
          </Wrapper>
        </Wrapper>
      );

      expect(container).toMatchSnapshot();
    });

    it('renders Wrapper with row and reverse', () => {
      const container = render(
        <Wrapper row reverse>
          <Wrapper baseHorizontal>
            <Text>Item 1</Text>
          </Wrapper>

          <Wrapper baseHorizontal>
            <Text>Item 2</Text>
          </Wrapper>
        </Wrapper>
      );

      expect(container).toMatchSnapshot();
    });

    it('renders Wrapper with reverse', () => {
      const container = render(
        <Wrapper reverse>
          <Wrapper baseHorizontal>
            <Text>Item 1</Text>
          </Wrapper>

          <Wrapper baseHorizontal>
            <Text>Item 2</Text>
          </Wrapper>
        </Wrapper>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
