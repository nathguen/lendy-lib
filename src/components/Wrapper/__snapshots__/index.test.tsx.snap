// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper flex options renders Wrapper with NO row 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 2
    </Text>
  </View>
</View>
`;

exports[`Wrapper flex options renders Wrapper with reverse 1`] = `
<View
  style={
    Object {
      "flexDirection": "column-reverse",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 2
    </Text>
  </View>
</View>
`;

exports[`Wrapper flex options renders Wrapper with row 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 2
    </Text>
  </View>
</View>
`;

exports[`Wrapper flex options renders Wrapper with row and reverse 1`] = `
<View
  style={
    Object {
      "flexDirection": "row-reverse",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 1
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#191919",
          "fontSize": 17,
          "fontWeight": "400",
          "lineHeight": 22,
        }
      }
    >
      Item 2
    </Text>
  </View>
</View>
`;

exports[`Wrapper renders Wrapper 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
/>
`;

exports[`Wrapper renders Wrapper with baseHorizontal 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "paddingHorizontal": 8,
    }
  }
/>
`;

exports[`Wrapper renders Wrapper with baseHorizontal and baseVertical 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "padding": 8,
    }
  }
/>
`;

exports[`Wrapper renders Wrapper with baseVertical 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "paddingVertical": 8,
    }
  }
/>
`;

exports[`Wrapper renders Wrapper with children 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#191919",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Wrapper renders Wrapper with screen 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "padding": 16,
    }
  }
/>
`;
